android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/shopping/ShoppingProductAnalysisResult.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.shopping
+
+/**
+ * Holds the result of the analysis of a shopping product.
+ */
+interface ShoppingProductAnalysisResult {
+
+    /**
+     * Product identifier (ASIN/SKU)
+     */
+    val productId: String
```

Comment:
This is the file in the `concept/engine`, so this is not the GV-specific area of the code. This is a generic interface, which you are supposed to implement if you are adding this functionality to a specific engine. To make this as generic as possible I left only the ID field which seems like the minimum essential thing.

In the GV-specific area I am implementing this interface and adding all our fields (it's the `engine/gecko/shopping/GeckoShoppingProductAnalysisResult.kt` file)

Reply:
My understanding is, we cannot simply use the GeckoView type, because this type should be present in the method 

```
abstract fun requestProductAnalysis(
  url: String, 
  onResult: (ShoppingProductAnalysisResult) -> Unit, 
  onException: (Throwable) -> Unit)

```

that we are defining in the concept Engine. The concept Engine is a general concept, so it cannot use Gecko-specific types, therefore, we need a concept type. I made it an concept interface which you are supposed to implement for your specific engine.

(I think this is how things work in AC)